### PR TITLE
chore(reth-execution-errors) no_std

### PR DESCRIPTION
**Pull request by martinezjorge**
Towards #9478 

Adds `no_std` support for `reth-execution-errors`

### CODE REVIEW

File: crates/evm/execution-errors/src/lib.rs [hunk 1]
```diff
@@ -212,3 +226,15 @@ impl InternalBlockExecutionError {
...
+    fn from(error: PruneSegmentError) -> Self {
+        Self::Pruning(error)
+    }
+}
+
+impl From<ProviderError> for InternalBlockExecutionError {
+    fn from(error: ProviderError) -> Self {
+        Self::LatestBlock(error)
+    }
+}
```
Comment (emhane) [inline crates/evm/execution-errors/src/lib.rs:240]: replace all the from impls with [`derive_more::From`](https://jeltef.github.io/derive_more/derive_more/from.html)

**Review by emhane — DISMISSED** [review]

File: crates/evm/execution-errors/src/lib.rs (hunk 1 above)
Comment (martinezjorge) [reply crates/evm/execution-errors/src/lib.rs]: done

**Review by mattsse — CHANGES_REQUESTED** [review]
last nit

File: crates/evm/execution-errors/src/lib.rs [hunk 2]
```diff
@@ -11,74 +11,76 @@
...
-#[cfg(not(feature = "std"))]
-use alloc::{boxed::Box, string::String};
+use derive_more::{Display, From};
 
 pub mod trie;
 pub use trie::*;
 
 /// Transaction validation errors
-#[derive(thiserror_no_std::Error, Debug, Clone, PartialEq, Eq)]
+#[derive(Clone, Debug, Display, Eq, From, PartialEq)]
```
[the remark this answers is not in the document]
Comment (martinezjorge) [reply crates/evm/execution-errors/src/lib.rs]: no problem, done

File: crates/evm/execution-errors/src/lib.rs [hunk 3]
```diff
@@ -172,14 +205,13 @@ impl BlockExecutionError {
 }
 
 /// Internal (i.e., not validation or consensus related) `BlockExecutor` Errors
-#[derive(thiserror_no_std::Error, Debug)]
+#[derive(Display, Debug, From)]
```
[the remark this answers is not in the document]
Comment (martinezjorge) [reply crates/evm/execution-errors/src/lib.rs]: done

**Review by shekhirin — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**